### PR TITLE
Add Blackfire

### PR DESCRIPTION
**Pull request by Mathieudewet**

### CODE REVIEW

**Comment by dunglas** [comment]
Because the Blackfire agent is not a free software, I would comment these lines by default.

File: api/helm/api/values.yaml [hunk 1]
```diff
@@ -70,6 +70,10 @@ postgresql:
   persistence:
     enabled: false
 
+blackfire:
+    blackfire:
```
Comment (vincentchalamon) [inline api/helm/api/values.yaml]:
```suggestion
    enabled: false
```

File: ci/deploy.sh [hunk 2]
```diff
@@ -38,6 +39,9 @@ helm upgrade --install --reset-values --wait --force --namespace=${NAMESPACE} --
     --set nginx.repository=${NGINX_REPOSITORY} \
     --set varnish.repository=${VARNISH_REPOSITORY} \
     --set secret=${APP_SECRET} \
+    --set blackfire.blackfire.server_id=${BLACKFIRE_SERVER_ID} \
```
Comment (vincentchalamon) [inline ci/deploy.sh]:
```suggestion
    --set blackfire.server_id=${BLACKFIRE_SERVER_ID} \
```

File: ci/deploy.sh [hunk 3]
```diff
@@ -38,6 +39,9 @@ helm upgrade --install --reset-values --wait --force --namespace=${NAMESPACE} --
...
+    --set blackfire.blackfire.server_token=${BLACKFIRE_SERVER_TOKEN} \
```
Comment (vincentchalamon) [inline ci/deploy.sh]:
```suggestion
    --set blackfire.server_token=${BLACKFIRE_SERVER_TOKEN} \
```

File: ci/deploy.sh [hunk 4]
```diff
@@ -38,6 +39,9 @@ helm upgrade --install --reset-values --wait --force --namespace=${NAMESPACE} --
...
+    --set blackfire.blackfire.enabled=${BLACKFIRE_ENABLED} \
```
Comment (vincentchalamon) [inline ci/deploy.sh]:
```suggestion
    --set blackfire.enabled=${BLACKFIRE_ENABLED} \
```

File: api/helm/api/values.yaml [hunk 5]
```diff
@@ -70,6 +70,10 @@ postgresql:
...
+        enabled: false
```
Comment (vincentchalamon) [inline api/helm/api/values.yaml]: Missing options `server_id` & `server_token`

File: .travis.yml [hunk 6]
```diff
@@ -26,10 +26,15 @@ env:
 # Check that all needed travis environment variables are set.
 # When working in dev branchs we don't want to override the master builds so we need to define à DEV_ADMIN_BUCKET and a DEV_CLIENT_BUCKET.
 # If you really want branch naming, you can instead create buckets on the fly using subdomains e.g  gs://${BRANCH}.you-domain.com
+# For your information, we set the BLACKFIRE_SERVER_ID and BLACKFIRE_SERVER_TOKEN values in the CI repository. This permits us to ship the blackfire disabled by default but enable it in online demo.
```
Comment (dunglas) [inline .travis.yml]:
```suggestion
# Set the BLACKFIRE_SERVER_ID and BLACKFIRE_SERVER_TOKEN to enable Blackfire.io support.
```

File: .travis.yml [hunk 7]
```diff
@@ -26,10 +26,15 @@ env:
...
 before_install:
   - nvm install node
   - ./ci/before_install.sh
   - source ./ci/.env
+  - |
+    if [[ ! -z "${BLACKFIRE_SERVER_ID}" && ! -z "${BLACKFIRE_SERVER_TOKEN}" ]]; then
```
Comment (dunglas) [inline .travis.yml]: Useless

File: ci/.env [hunk 8]
```diff
@@ -8,3 +8,4 @@ export MULTI_BRANCH=0
 export REPOSITORY='api-platform/demo'
 export DEPLOYMENT_BRANCH='master'
 export PROD_DNS='demo.api-platform.com'
+export BLACKFIRE_ENABLED=false #Default set to false as it's a non free solution
```
Comment (dunglas) [inline ci/.env]: useless

File: ci/before_deploy.sh [hunk 9]
```diff
@@ -10,4 +10,4 @@ gcloud auth activate-service-account ${CI_SERVICE_ACCOUNT} --key-file travis-ser
 gcloud config set compute/zone europe-west1-c
 gcloud config set core/project ${PROJECT_ID}
 gcloud container clusters get-credentials api-platform-demo --zone europe-west1-c --project ${PROJECT_ID}
-helm init --upgrade
+helm init --force-upgrade
```
Comment (dunglas) [inline ci/before_deploy.sh]: Why?

Comment (Mathieudewet) [reply ci/before_deploy.sh]: Because tiller server version is newer than client one.

File: ci/.env [hunk 10]
```diff
@@ -6,5 +6,5 @@
 export CURRENT_CI='travis'
 export MULTI_BRANCH=0
 export REPOSITORY='api-platform/demo'
-export DEPLOYMENT_BRANCH='master'
+export DEPLOYMENT_BRANCH='feature/blackfire'
```
Comment (vincentchalamon) [inline ci/.env]:
```suggestion
export DEPLOYMENT_BRANCH='master'
```

Comment (Mathieudewet) [reply ci/.env]: This is in order to allow travis to deploy testing branch during dev.
I'll reset it to master when ready to merge.

**Comment by vincentchalamon** [comment]
Thanks @Mathieudewet

**Pull request merged**